docs/architecture-overview.md

```diff
@@ -0,0 +1,124 @@
+
+- description of individual object types
+  - for each one:
+    - description
+      - as much as possible, link to kdocs rather than duplicating
+      - some additional information about how they fit into the larger whole
+    - this object reacts to updates from X
+    - this object notifies Y of changes
+    - where valuable, relationships (1:1, 1:1+, 1:many)
+  - objects
+    - Store
+      - see AC docs
+      - store:fragment == 1:1
+      - store:reducer == 1:1
+      - only one will ever be alive at one time
+    - Action
+      - see AC docs
+    - State
+      - see AC docs
+      - immutable
+      - as much as possible, 1:1 relationship with UI
+        - explain why
+        - stress that _this makes testing easier_
+    - Reducer
+      - see AC docs
+      - must be a pure function
+      - where the bulk of 'logic' should live
+    - Controller
+      - when called by Interactor, collects context and does one (and _only_ one) of the following:
+        - send Action to Store
+        - route to new Fragment
+          - optionally including data passed as intent extras
+    - Interactor
+      - fragment:interactor == 1:1+
+      - interactor:controller == 1:0+
+      - interactor:interactor == 1:0+
+      - delegates user actions to Controllers or other Interactors
+      - this layer exists to
+        - keep the View from having to know what objects are listening to it
+        - keep controllers from having to handle routing calls to correct objects + creating Actions
+    - View
+      - fragment:view == 1:1+
+      - view:interactor == 1:1
+      - defines mapping of state -> UI
+      - updates views whenever a new state obj is received
+  
+
+- controller changes can happen either as an update to the store, or by routing to a new fragment + store
+- communication between stores only happens via transactions 
+- stores & reducers are persisted across configuration changes (via Arch Components VM), but created / destroyed during fragment transactions
+  - everything else is destroyed / recreated during configuration changes 
+  
+
+
+## Simplified Example
+
+(this might be overkill / out of scope? reading through a fully fleshed out section of the code base is hard. might 
+be good to include this in order to onboard contributors and new devs in a simpler context)
+
+- simple, drawn example
+- text message app
+  - contact list screen + conversation screen
+  - when on contact screen:
+    - updating a contact changes the state
+    - selecting a conversation routes to a new screen
+  - when on conversation screen
+    - backing out routes to a new screen + sends an extra to update 'unread' status for the conversation
+     
+    
+    
+## Differences from previous architecture
+
+- same basic concept
+  - makes the migration much easier
+- action -> change was very flexible, but usually unnecessary
+  - added complexity
+  - distinction was removed
+- Rx 
+  - slowed startup
```

Comment:
Coroutines were actually worse until we implemented a fix in R8. Coroutines Dispatchers.Main blocked the main thread >200ms. Rx AndroidSchedulers.MainThread blocked the main thread 42ms on the same hardware. Coroutines are now faster because we strip out the wasted, blocking code using an early version of R8.

Reply:
Good context, thank you